packages/unuse/src/useToggle/index.ts

```diff
@@ -0,0 +1,81 @@
+import { toUnSignal } from '../toUnSignal';
+import type { MaybeUnRef, UnRef } from '../unAccess';
+import { isUnRef, unAccess } from '../unAccess';
+import type { ReadonlyUnResolveReturn } from '../unResolve';
+import { unResolve } from '../unResolve';
+
+export type ToggleFn<
+  TTruthy = true,
+  TFalsy = false,
+  T extends TTruthy | TFalsy | undefined = TTruthy | TFalsy | undefined,
+> = (value?: T) => T;
+
+export type UseToggleReturn<
+  TTruthy = true,
+  TFalsy = false,
+  T extends TTruthy | TFalsy | undefined = TTruthy | TFalsy | undefined,
+> =
+  | [ReadonlyUnResolveReturn<T>, ToggleFn<TTruthy, TFalsy, T>]
+  | ToggleFn<TTruthy, TFalsy, T>;
+
+export interface UseToggleOptions<TTruthy = true, TFalsy = false> {
+  truthyValue?: MaybeUnRef<TTruthy>;
+  falsyValue?: MaybeUnRef<TFalsy>;
+}
+
+export function useToggle<
+  TTruthy = true,
+  TFalsy = false,
+  T extends TTruthy | TFalsy = TTruthy | TFalsy,
+>(
+  initialValue: UnRef<T>,
+  options?: UseToggleOptions<TTruthy, TFalsy>
+): ToggleFn<TTruthy, TFalsy, T>;
+export function useToggle<
+  TTruthy = true,
+  TFalsy = false,
+  T extends TTruthy | TFalsy = TTruthy | TFalsy,
+>(
+  initialValue?: T,
```

Comment:
Do we have a convention for the name, it should be `defaultValue` I think? And the type could be `T | UseToggleOptions` considering it's optional?

Reply:
No, you confusing potential something's here.

`options` will always we the second argument (for useToggle), as either the first value is fully omitted and therefore false, or it is a MaybeRef<boolean> or it is a non boolean but then it has options.

So maybe I need to rewrite the overloads to reflect this.

Reply:
I think `useToggle('off', { falsyValue: 'off' })` is strange, it should just be simpler `useToggle({ falsyValue: 'off' })`, (the `defaultValue` is `falsyValue` as always by default)

Reply:
I think VueUse did it in https://vueuse.org/shared/useToggle/#usetoggle as the `initialValue` can be a reactive value AND also in your case you would also need to add a `truthyValue` as otherwise `useToggle` can't know what the `truthyValue` value should be. I hope it is not `{ falsyValue: 'off', truthyValue: true }` as this would be really confusing tbh.

Reply:
I omitted `truthyValue` because it's not the key point.

I think we can at least make the `falsyValue = intialValue = false`, so `falseValue` continue to be optinal.

Reply:
_what in the world_ 🫠 nope, sry XD
lets not do that, please

Reply:
Why is

```js
const toggle = useToggle('OFF', {
  truthyValue: status,
});
```

wrose than

```js
const toggle = useToggle('OFF', {
  truthyValue: status,
  falsyValue: 'OFF',
});
```

The only difference is that I omitted optional `falsyValue`? Is that a huge difference with VueUse?

Reply:
I was just saying omitting `falsyValue` should work, and a `MaybeRef` `initialValue` should also work at the same time.

```js
const value = unSignal('OFF')

const toggle = useToggle(value, {
  truthyValue: status,
});
```